File: snuggle_server/src/main/java/com/ssafy/snuggle/model/service/LikeService.java
package com.ssafy.snuggle.model.service;

import java.util.List;

import com.ssafy.snuggle.model.dto.Like;
import com.ssafy.snuggle.model.dto.Product;

public interface LikeService {

	/**
	 * Like을 등록한다. 사용자가 상품에서 좋아요 눌렀을 때
	 * 
	 * @param Like
	 */
	int addLike(Like favorite);

	/**
	 * 사용자가 좋아요 한 product 리스트 조회
	 * 
	 * @param userId
	 */
	List<Product> getLikeProductList(String userId);

}

File: snuggle_final_app/app/src/main/java/com/ssafy/snuggle_final_app/data/service/LikeService.kt
package com.ssafy.snuggle_final_app.data.service

import com.ssafy.snuggle_final_app.data.model.dto.Like
import com.ssafy.snuggle_final_app.data.model.dto.Product
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface LikeService {

    @POST("/snuggle/like/insert")
    suspend fun addLike(@Body like: Like): Boolean

    @GET("/snuggle/like/info/{userId}")
    suspend fun getLikeListByUser(@Path("userId") userId: String): List<Product>


}